core/src/bson_type_info.rs

```diff
@@ -560,6 +585,29 @@ impl BsonTypeInfo {
         column_size: make_default_attr_func!(Some(23)),
         simple_type_info: None,
     };
+    pub const DATETIME: BsonTypeInfo = BsonTypeInfo {
```

Comment:
SQL_DATETIME is not special SQL type.
This is only used for the verbose type name, which in conjunction with the sql code gives you the final type.
See here : https://learn.microsoft.com/en-us/sql/odbc/reference/appendixes/data-type-identifiers-and-descriptors?view=sql-server-ver16

Reply:
This standard is just incredibly messy